admin/services/data-access/pupil.data.service.js

```diff
@@ -59,4 +59,23 @@ pupilDataService.update = async function (id, doc) {
   })
 }
 
+/**
+ * Returns pupils filtered by school and sorted by field and direction (asc/desc)
+ * @param schoolId
+ * @param sortingField
+ * @param sortingDirection
+ * @returns {Array}
+ */
+pupilDataService.getSortedPupils = async (schoolId, sortingField, sortingDirection) => {
+  // TODO: Introduce integration tests
+  const sort = {}
+  sort[sortingField] = sortingDirection
+  const pupils = await Pupil
```

Comment:
As this is an async service, which means the caller *must* await on the service for the resolve(), you can just write it like this:

```
return Pupil
.find()
.lean()
.exec()
```

Reply:
There shorter form would be `return await ...` but you can lose the await as the caller is await'ing anyway.

Reply:
changed 👍